### PR TITLE
Feature/sep6 72 remove token validation for endpoints that do not require it

### PR DESCRIPTION
**Pull request by KhaledHammoun**

### CODE REVIEW

File: backend/src/main/java/com/sep6/backend/security/config/SecurityConfiguration.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .authorizeHttpRequests()
                 .requestMatchers(
                         "/accounts/**",
-                        "/movies/*/reviews/")
+                        "/movies/*/reviews")
```
Comment (tomondre) [inline backend/src/main/java/com/sep6/backend/security/config/SecurityConfiguration.java:46]: What about /movie/{movieId}/reviews/{reviewId}? I believe  we will need to include also another request matcher for /movies/*/reviews/*

File: backend/src/main/java/com/sep6/backend/security/config/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ protected void doFilterInternal(
             @NonNull FilterChain filterChain
     ) throws ServletException, IOException {
         log.info("Filtering request: {}", request.getServletPath());
-        if (request.getServletPath().contains("/auth")) {
+        if ((!request.getServletPath().matches("/accounts/.*")) || (!request.getServletPath().matches("/movies/[^/]+/reviews"))) {
```
Comment (tomondre) [inline backend/src/main/java/com/sep6/backend/security/config/JwtAuthenticationFilter.java:38]: What is the if statement for? :smile:

File: backend/src/main/java/com/sep6/backend/security/config/SecurityConfiguration.java (hunk 1 above)
Comment (KhaledHammoun) [reply backend/src/main/java/com/sep6/backend/security/config/SecurityConfiguration.java:46]: okay, then it should be /movies/*/reviews/**

File: backend/src/main/java/com/sep6/backend/security/config/JwtAuthenticationFilter.java (hunk 2 above)
Comment (KhaledHammoun) [reply backend/src/main/java/com/sep6/backend/security/config/JwtAuthenticationFilter.java]: It is so we do not validate the token on endpoints that do not need authentication

File: backend/src/main/java/com/sep6/backend/security/config/SecurityConfiguration.java (hunk 1 above)
Comment (KhaledHammoun) [reply backend/src/main/java/com/sep6/backend/security/config/SecurityConfiguration.java]: Done

**Review by tomondre — APPROVED** [review]

**Review by tomondre — APPROVED** [review]

**Pull request merged**